base/reinterpretarray.jl

```diff
@@ -172,6 +172,10 @@ end
 
 function strides(a::ReinterpretArray)
     a.parent isa StridedArray || throw(ArgumentError("Parent must be strided."))
+    if a isa ReshapedReinterpretArray && elsize(a) <= elsize(parent(a))
+        N = elsize(parent(a)) ÷ elsize(a)
+        return (1, map(i->N*i, strides(a.parent)...))
```

Comment:
Sorry for my carelessness:
```suggestion
        return (1, map(i->N*i, strides(a.parent))...)
```
Test passed because we only use it for `Vector` case.
So maybe we need some test for `Matrix` inputs.
And it would be good to add non-1 stride test to LinearAlgebra

Reply:
I'll add more tests. IIUC, are you suggesting adding tests for `Matrix{T} * Matrix{Complex{T}} where {T<:BlasFloat}`?

Reply:
I mean add some `strides` test for `ReshapedReinterpretArray` with `Matrix`-like parent.

On `LinearAlgebra` side I think we'd better add test like `v3 = @view v[end:-1:begin]; M2t * v3`. (To make sure we pass the correct stride/pointer information to BLAS)

Reply:
I've added tests for non-contiguous matrices. I assume you're adding tests for strides in #44027?

Reply:
That's a more complete fix.
It's OK if the reinterpreting based fallback is fully tested. (Would be another good test suite for #44027)